docs/features/interceptors.md

```diff
@@ -178,6 +207,8 @@ Although interceptors are an experimental feature, there will be no explicit opt
 
 PROTOTYPE(ic): The BCL might not ship the attributes required by this feature, instead requiring them to be declared in some library brought in by a package reference, or in the user's project. But we haven't confirmed this.
 
+PROTOTYPE(ic): Feedback from Jared is that we need to have an opt-in step, even if it is only performed by generators, so that any teams adopting this understand the level of support they can expect for this. Most likely, we would do this by checking for `/features=interceptors` on the command line (`<Features>interceptors</Features>` in msbuild). Then generators that want to use interceptors would just ship a bit of msbuild logic to ensure the appropriate feature flag is being passed to the compiler.
```

Comment:
I think this is important **if** we end up labeling this as experimental for .NET 8. There will absolutely be generators that take advantage of this setting. I don't want to fall into a trap where we label it experimental, have no opt-in mechanism and generators using interceptors reach a critical mass such that we can't change it. Just as customers had to opt into experimental features like DIM, they need to opt into this. 

At the same time, I feel like for scenarios like ASP.NET AOT this opt in is implicit. The props / targets in SDK are free to implicitly set this flag such that we have a friction free experience there. Part of deciding this is experimental would be ASP.NET agreeing to adjust the generator to new compiler behavior should we change things. Essentially we're agreeing to own keeping this overall scenario working as a team even if the implementation details change a bit. 

Note: this is all still predicated on whether this is labeled experimental or not in .NET 8.